soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,178 @@
+import skills.move
+import skills.capture
+
+
+class distraction(standard_play.StandardPlay):
+
+    class State (enum.Enum):
+        setup = 1, 'capture ball and move distractor and striker into position'
+        optional_adjustment = 2, 'setup a pass to the center right'
+        cetner_pass = 3, 'pass to the center right of the field'
+        passing = 4, 'pass from first capture to distractor'
+        cross = 5, 'pass from distractor to striker'
+        shoot = 6, 'shoot at goal'
+    
+    def __init__(self):
+        super().__init__(continuous=False)  
+
+        self.ball_is_far = False
+        self.distraction_point = robocup.Point(0.40*constants.Field.Width, 0.95*constants.Field.Length) #the first distraction point
+        self.distraction_recieve_pass_point = robocup.Point(0.40*constants.Field.Width, 0.8*constants.Field.Length) #the second distraction point
+        self.striker_point = robocup.Point(-0.40*constants.Field.Width, 0.9*constants.Field.Length) #striker's position
+        self.center = robocup.Point(0.5*constants.Field.Width,0.5*constants.Field.Length) #center of field position, used if ball is far
+        self.distract_box_min = robocup.Point(0.33*constants.Field.Width/2,(7/9)*constants.Field.Length)
+        self.distract_box_max = robocup.Point(constants.Field.Width/2, constants.Field.Length)
+        self.striker_box_min = robocup.Point(-0.33*constants.Field.Width/2,(7/9)*constants.Field.Length)
+        self.striker_box_max = robocup.Point(-constants.Field.Width/2, constants.Field.Length)
+
+        states = [distraction.State.setup, distraction.State.optional_adjustment, distraction.State.cetner_pass,
+         distraction.State.passing, distraction.State.cross, distraction.State.shoot]
+
+        for i in states:
+            self.add_state(i, behavior.Behavior.State.running)
+
+        self.add_transition(behavior.Behavior.State.start,
+                        distraction.State.setup, 
+                        lambda: True, 
+                        'immediately')
+
+        self.add_transition(distraction.State.setup,
+                        distraction.State.passing, 
+                        lambda: not self.ball_is_far,
+                        'capture the ball and pass to a distracting robot if the ball is not far away')
+
+        #optional_adjustment can be skipped if ball is not far from goal
+        self.add_transition(distraction.State.setup, 
+                        distraction.State.optional_adjustment, 
+                        lambda: self.ball_is_far,
+                        'if the ball is far away, then go from capturing the ball to getting robots set up to recieve a closer pass')
+
+        self.add_transition(distraction.State.optional_adjustment,
+                        distraction.State.cetner_pass, 
+                        lambda: self.subbehavior_with_name('move half').is_done_running() and self.subbehavior_with_name('make striker stay').is_done_running() , 
```

Comment:
This just waits for the two robots to move to their point. We may not have the ball still